x/auth/signing/cache.go

```diff
@@ -0,0 +1,89 @@
+package signing
+
+import (
+	"sync"
+
+	lru "github.com/hashicorp/golang-lru/v2"
+)
+
+const (
+	TxHashLen        = 32
+	AddressStringLen = 2 + 20*2
+)
+
+var initSigCacheOnce sync.Once
+
+// Cache is a cache of verified signatures
+type Cache struct {
+	data *lru.Cache[string, []byte]
+}
+
+// NewSignatureCache initializes the signature cache.
+// signature verification is one of the most expensive parts in verification
+// by caching it we avoid needing to verify the same signature multiple times
+// This is only initialized once.
+func NewSignatureCache() *Cache {
```

Comment:
So if `NewSignatureCache` is called twice, on the 2nd call, we'll return an empty `&Cache{}`. Is this what we expect?